### PR TITLE
Instructor: add course: Wrong redirection after adding a course #1168

### LINKED ISSUE #1168: Instructor: add course: Wrong redirection after adding a course
_From [victorph...@gmail.com](https://code.google.com/u/114670200553077300926/) on August 28, 2013 19:58:20_

after adding a course, refreshing the web page display the error "A course by the same ID already exists in the system" instead of actually reloading the add new course page

_Original issue: http://code.google.com/p/teammatespes/issues/detail?id=1169_

### PR DESCRIPTION
**Pull request by Mynk96**
Fixes #1168 
**Outline of Solution**
Used redirectResult after successfully adding a course.

### CODE REVIEW

**Comment by Mynk96** [comment]
ready for review!
should I change 'redirectResult' variable name to 'redirectedResult' for better understanding or is it just fine?

**Comment by Mynk96** [comment]
@whipermr5 please review.

File: src/test/java/teammates/test/cases/action/InstructorCourseAddActionTest.java [hunk 1]
```diff
@@ -71,10 +75,10 @@ public void testExecuteAndPostProcess() throws Exception {
         addAction = getAction(Const.ParamsNames.COURSE_ID, "ticac.tpa1.id",
                               Const.ParamsNames.COURSE_NAME, "ticac tpa1 name",
                               Const.ParamsNames.COURSE_TIME_ZONE, "UTC");
-        pageResult = getShowPageResult(addAction);
+        RedirectResult redirectResult = getRedirectResult(addAction);
 
-        pageData = (InstructorCoursesPageData) pageResult.data;
-        assertEquals(2, pageData.getActiveCourses().getRows().size() + pageData.getArchivedCourses().getRows().size());
+        List<CourseAttributes> courseList = CoursesLogic.inst().getCoursesForInstructor(instructorId);
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/action/InstructorCourseAddActionTest.java]: A question: are such direct references to Logic used in any other action tests?

Comment (Mynk96) [inline src/test/java/teammates/test/cases/action/InstructorCourseAddActionTest.java]: Yes, for instance in instructurDeleteActionTest.

**Pull request merged**